src/lib/MediaPreview/MediaPreview.vue

```diff
@@ -67,26 +108,48 @@ export default {
   },
 
   props: {
-    file: { type: Object, required: true }
+    files: { type: Array, required: true },
+    index: { type: Number, required: true }
   },
 
   emits: ['close-media-preview'],
 
   data() {
     return {
-      isLoadingIframe: true
+      isLoadingIframe: true,
+      file: this.files[this.index],
+      showArrows: this.files.length > 1,
+      fileIndex: this.index,
+      fileContent: null,
+      cachedFiles: {},
+      isFetchingFile: false,
+      isSVGLoading: true
     }
   },
 
   computed: {
     isImage() {
       return isImageFile(this.file)
     },
+    isSVG() {
+      const temp = isSVGFile(this.file)
+      if (temp) {
+        this.loadSVG(this.file)
+      }
+      return temp
+    },
     isVideo() {
       return isVideoFile(this.file)
     },
     isPdf() {
       return isPdfFile(this.file)
+    },
+    isText() {
+      const temp = isTextFile(this.file)
+      if (temp) {
+        this.getFileContent(this.file)
```

Comment:
Olhando o código pela primeira vez, eu tive que pensar um pouco para entender o que estava acontecendo aqui. Eu imaginei que alguma coisa que cache estava acontecendo, dai tive que olhar o `getFileContent` para ter certeza. Vamos evitar esse trabalho para alguém no futuro.

Podemos fazer algo como:

```suggestion
        this.loadAndCacheFileContent(this.file)
```

Dai o método `loadAndCacheFileContent()` você pode só chamar o `getFileContent()`, mas pelo menos a lógica e a semântica do código ficam mais fáceis. E, de quebra, no futuro podemos fazer uma lógica separada para `loadAndCacheFileContent()` se for necessário.

Reply:
Realmente a lógica ficou um pouco confusa, extraí a responsabilidade pelo `cache` e pelo `download` em funções separadas, obrigado pela sugestão.